### DISCUSSION TITLE
Cannot implicitly convert type 'NetTopologySuite.Geometries.Coordinate' to 'DotSpatial.Topology.Coordinate'

### DISCUSSION THREAD

**Discussion by abt5555** [DotSpatial/DotSpatial#1511] [Q&A] [answered]
how to solve 

Cannot implicitly convert type 'NetTopologySuite.Geometries.Coordinate' to 'DotSpatial.Topology.Coordinate' error

any help? dotspatial 4.0 visual studio 2022

**Comment by jany-tenaj** [comment]
DotSpatial.Topology isn't used in DotSpatial 4.0. So explain more clearly what you are doing to get this error.

**Reply by abt5555** [reply]
Thanks for the reply. I want to create points, lines and polygon area on map with mouse clicks.  the part of code in below, I am getting error

`Coordinate coord = map1.PixelToProj(e.Location);`

I solved it for creating points with below code

```
                    if ((pointmouseClick))
                        {

                            NetTopologySuite.Geometries.Coordinate netCoordinate = map1.PixelToProj(e.Location);

                            DotSpatial.Topology.Coordinate dotCoordinate = new DotSpatial.Topology.Coordinate(netCoordinate.X, netCoordinate.Y);

                            DotSpatial.Topology.Point point = new DotSpatial.Topology.Point(dotCoordinate);

                            NetTopologySuite.Geometries.Coordinate netpoint = new NetTopologySuite.Geometries.Coordinate(point.X, point.Y);

                            NetTopologySuite.Geometries.GeometryFactory geometryFactory = new NetTopologySuite.Geometries.GeometryFactory();
                            NetTopologySuite.Geometries.Point sonnetpoint = geometryFactory.CreatePoint(netpoint);

                            IFeature currentFeature = pointF.AddFeature(sonnetpoint);

                            pointID = pointID + 1;

                            currentFeature.DataRow["PointID"] = pointID;

                            map1.ResetBuffer();

                        }
```

If there is easier way please let me know

**Comment by jany-tenaj** [comment, marked as answer]
Your source code is more than strange. You're mixing 4.0 and 1.x code.

Make sure you're only using DotSpatial 4.0 from nuget. 

Because you are using e.Location I think your code comes from a OnMouseDown / OnMouseUp event, that has a GeoMouseArgs parameter. 

If you have the correct dlls from nuget you should be able to add your point inside a OnMouseDown / OnMouseUp event with the following code:

```
if (pointmouseClick)
                        {
                            Feature currentFeature = pointF.AddFeature(e.GeographicLocation);

                            pointID = pointID + 1;

                            currentFeature.DataRow["PointID"] = pointID;

                            map1.ResetBuffer();

                        }
```

Be aware of #801.

**Reply by abt5555** [reply]
thank you so much Jany I am a newbie

**Reply by dvDevel** [reply]
Thank you @jany-tenaj 
 I'm also looking for the same thing with @abt5555.
I believe the event you mentioned is GeoMouseMove because I have implemented all the event methods in the properties view related to the DotSpatial.Controls.Map.

Unfortunately, the pointF.AddFeature() method requires a parameter of Geometry type.
 I just wanted to confirm if I understood you correctly.

Thank you again.

**Reply by jany-tenaj** [reply]
e.GeographicLocation is a Point which is a Geometry.

**Reply by dvDevel** [reply]
Thank you @jany-tenaj 
In my case e.GeographicLocation returns the Coordinate type. 
 
 map1_GeoMouseMove(object sender, GeoMouseArgs e)

I think what I missed

**Comment by abt5555** [comment]
for working dotspatial 1.7, which version of visual studio and net should I use? I am following tutorial from [https://baixardoc.com/preview/dotspatial-tutorial-1-5ceeebcb0111c](url) I think they were prepared for dotspatial 1.7

**Comment by jany-tenaj** [comment]
Don't use version 1.7. It's obsolete. If you need help explain clearly what you want to do and we'll see whether we can give you a current example code.

**Reply by abt5555** [reply]
Thank you Jany
